helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java

```diff
@@ -191,8 +190,7 @@ public HelixTaskExecutor(ParticipantStatusMonitor participantStatusMonitor,
     _batchMessageExecutorService = Executors.newCachedThreadPool();
     _monitor.createExecutorMonitor("BatchMessageExecutor", _batchMessageExecutorService);
 
-    _resourcesThreadpoolChecked = Collections.newSetFromMap(new ConcurrentHashMap<>());
-    _transitionTypeThreadpoolChecked = Collections.newSetFromMap(new ConcurrentHashMap<>());
+    _threadpoolChecked = Collections.newSetFromMap(new ConcurrentHashMap<>());
```

Comment:
How you can combine them? Say we just need the "resource" based thread pool, are you doing prefixing searching?


This is already changing the logic.

Reply:
Previously we use different sets to do search for different property-based threadpool. Right now I'm combining them into one set, but making the key generation structural, so we maintain the capability of searching threadpool in different levels.

Reply:
I read through the code. It breaks the logic... Let's have some chat about it.

Reply:
Synced offline. It is not logic change but code style could be hard to read. Let's refactor it with minimum changes.